manticore/core/state.py

```diff
@@ -196,7 +201,7 @@ def abandon(self):
 
         Note: This must be called from the Executor loop, or a :func:`~manticore.Manticore.hook`.
         '''
-        raise TerminateState("Abandoned state")
+        raise AbandonState
```

Comment:
`AbandonState` inherits the `__init__` method from `TerminateState`, so this line misses the required string argument and hence will throw a different exception. One way to fix this issue would be to give `AbandonState` its own `__init__` method with a message argument that has the default value `'Abandoned state'`.

Reply:
:+1:

```python
In [1]: class A(Exception): pass

In [2]: class B(A):
   ...:     def __init__(self, msg):
   ...:         pass
   ...:

In [3]: class C(B): pass

In [4]:

In [4]: raise C
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-4-4a531f008761> in <module>()
----> 1 raise C

TypeError: __init__() missing 1 required positional argument: 'msg'
```